vcd/resource_vcd_external_network.go

```diff
@@ -310,3 +340,110 @@ func GetVcenterHref(vcdClient *govcd.VCDClient, name string) (string, error) {
 	}
 	return virtualCenters[0].HREF, nil
 }
+
+// Retrieves an external network and returns an interface map corresponding to the resource
+// Input: vcdClient , external network Identifier (either name or ID)
+// output: StringMap (representing the resource), external network ID, error
+func getExternalNetworkResource(vcdClient *govcd.VCDClient, extNetIdentifier string) (StringMap, string, error) {
+	var extNetRes = make(StringMap)
+
+	externalNetwork, err := vcdClient.GetExternalNetworkByNameOrId(extNetIdentifier)
+	if err != nil {
+		return nil, "", fmt.Errorf("error fetching external network details %s", err)
+	}
+
+	// Although the resource allows an array of vCenters,
+	// the current implementation of external network only records one.
+	vcenterHref := ""
+
+	if externalNetwork.ExternalNetwork.VimPortGroupRef != nil && externalNetwork.ExternalNetwork.VimPortGroupRef.VimServerRef != nil {
+		vcenterHref = externalNetwork.ExternalNetwork.VimPortGroupRef.VimServerRef.HREF
+	} else {
+		return nil, "", fmt.Errorf("error retrieving VC HREF : %s", err)
+	}
+	virtualCenters, err := govcd.QueryVirtualCenters(vcdClient, fmt.Sprintf("(href==%s)", vcenterHref))
+	if err != nil {
+		return nil, "", err
+	}
+	if len(virtualCenters) == 0 {
+		return nil, "", fmt.Errorf("no virtual centers found with HREF %s", vcenterHref)
+	}
+
+	var ipScopes []StringMap
+	for _, ips := range externalNetwork.ExternalNetwork.Configuration.IPScopes.IPScope {
```

Comment:
We haven't, and given the type of vCDs that we use for testing, we could do it only if we created an external network when the server is still empty, and it could not be in the test suite.

Reply:
It works. I've tried to run `terraform plan` 300 times on a resource with multiple ip scopes and static pool definitions and it always worked (never got it in the odd order)